src/runtime/contrib/ethosn/ethosn_runtime.cc

```diff
@@ -0,0 +1,127 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ * \file ethosn_runtime.cc
+ * \brief Execution handling of Ethos-N command streams.
+ */
+
+#include "ethosn_runtime.h"
+
+#include <dmlc/memory_io.h>
+#include <tvm/runtime/c_runtime_api.h>
+#include <tvm/runtime/memory.h>
+#include <tvm/runtime/module.h>
+#include <tvm/runtime/object.h>
+#include <tvm/runtime/packed_func.h>
+#include <tvm/runtime/registry.h>
+
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "../../file_util.h"
+#include "ethosn_device.h"
+#include "ethosn_driver_library/Inference.hpp"
+#include "ethosn_driver_library/Network.hpp"
+#include "ethosn_support_library/Support.hpp"
+
+namespace tvm {
+namespace runtime {
+namespace ethosn {
+
+namespace sl = ::ethosn::support_library;
+namespace dl = ::ethosn::driver_library;
+
+EthosnModule::EthosnModule(std::vector<OrderedCompiledNetwork>* cmms) {
+  for (auto& it : *cmms) {
+    network_map_[it.name].name = it.name;
+    network_map_[it.name].cmm = std::move(it.cmm);
+    network_map_[it.name].inputs = it.inputs;
+    network_map_[it.name].outputs = it.outputs;
+  }
+}
+
+PackedFunc EthosnModule::GetFunction(const std::string& name,
+                                     const ObjectPtr<Object>& sptr_to_self) {
+  if (network_map_.find(name) != network_map_.end()) {
+    return PackedFunc([sptr_to_self, this, name](TVMArgs args, TVMRetValue* rv) {
+      *rv = Inference(args, network_map_[name].cmm.get(), network_map_[name].inputs,
```

Comment:
A very minor issue, but isn't it looking up `network_map_` on every inference call?

Reply:
It is, to get the correct function from the module according to the name.